libraries/transformer/src/main/java/androidx/media3/transformer/FrameworkMuxer.java

```diff
@@ -276,9 +284,64 @@ private static ImmutableList<String> getSupportedVideoSampleMimeTypes() {
     if (SDK_INT >= 24) {
       supportedMimeTypes.add(MimeTypes.VIDEO_H265);
     }
+    if (SDK_INT >= 33) {
+      supportedMimeTypes.add(MimeTypes.VIDEO_DOLBY_VISION);
+    }
     if (SDK_INT >= 34) {
       supportedMimeTypes.add(MimeTypes.VIDEO_AV1);
     }
     return supportedMimeTypes.build();
   }
+
+  // Get Dolby Vision profile
+  // Refer to https://professionalsupport.dolby.com/s/article/What-is-Dolby-Vision-Profile
+  @RequiresApi(api = Build.VERSION_CODES.TIRAMISU)
+  private static int getDvProfile(Format format) {
+    // Currently, only profile 8 is supported for encoding
+    // TODO: set profile ID based on format.
+    return MediaCodecInfo.CodecProfileLevel.DolbyVisionProfileDvheSt;
+  }
+
+  // Get Dolby Vision level
+  // Refer to https://professionalsupport.dolby.com/s/article/What-is-Dolby-Vision-Profile
+  @RequiresApi(api = Build.VERSION_CODES.TIRAMISU)
+  private static int getDvLevel(Format format) {
+    int level = -1;
+    int maxWidthHeight = Math.max(format.width, format.height);
+    float pps = format.width * format.height * format.frameRate;
+
+    if (maxWidthHeight <= 1280) {
+      if (pps <= 22118400) {
+        level = MediaCodecInfo.CodecProfileLevel.DolbyVisionLevelHd24;  // Level 01
+      } else { // pps <= 27648000
+        level = MediaCodecInfo.CodecProfileLevel.DolbyVisionLevelHd30;  // Level 02
+      }
+    } else if (maxWidthHeight <= 1920 && pps <= 49766400) {
+      level = MediaCodecInfo.CodecProfileLevel.DolbyVisionLevelFhd24;  // Level 03
+    } else if (maxWidthHeight <= 2560 && pps <= 62208000) {
+      level = MediaCodecInfo.CodecProfileLevel.DolbyVisionLevelFhd30;  // Level 04
+    } else if (maxWidthHeight <= 3840) {
```

Comment:
Can we use this existing method?

https://source.corp.google.com/piper///depot/google3/third_party/java_src/android_libs/media/libraries/exoplayer/src/main/java/androidx/media3/exoplayer/mediacodec/MediaCodecUtil.java;l=307?q=MediaCodecUti#:~:text=306-,307,-308

Reply:
It's for different purpose. The method in your link is to PARSE the Dolby Vision profile and level from codec string. The use case is to play a streaming content. You can get codec string from a MPEG-DASH manifest file. Then paring this codec string, you will get profile/level info. This info can be used to select which Dolby Vision decoder to handle this streaming content (H.264 based Dolby Vision decoder or H.265 based Dolby Vision decoder). In this case, Format.codecs is a valid codec string.

As far as this code contribution case, it is used to decide which Dolby Vision profile/level should be used based on incoming content width/height/framerate. In this case. Format.codecs is NULL.